Task 4/Task4/Task4/CastomGit.cs

```diff
@@ -0,0 +1,150 @@
+
+        private UserCommunication communicator = new UserCommunication();
+
+        private DateTime LogTime;
+
+        public CastomGit(string path)
+        {
+            this.path = path;
+        }
+
+        
+        public void StartProgram()
+        {
+            string[] files = Directory.GetFiles(path, "*.txt");
+
+            DateTime currentTime = DateTime.Now;
+
+            string initialCommitPath = path + "\\" + currentTime.ToString("dd.MM.yyyy.HH.mm.ss");
+
+            Directory.CreateDirectory(initialCommitPath);
+
+            for (int i = 0; i< files.Length; i++)
+            {
+                String text = File.ReadAllText(files[i]);
+
+                File.WriteAllText(initialCommitPath + files[i].Remove(0, path.Length), text);
+
+                FilePastPath.Add(files[i], initialCommitPath + files[i].Remove(0, path.Length));
+            }
+
+            Tracking();
+
+        }
+
+
+        public void Tracking()
+        {
+            Console.WriteLine("Tracking...");
+
+            using FileSystemWatcher fileWatcher = new FileSystemWatcher(path);
+
+            fileWatcher.NotifyFilter = NotifyFilters.Attributes
+                     | NotifyFilters.CreationTime
+                     | NotifyFilters.DirectoryName
+                     | NotifyFilters.FileName
+                     | NotifyFilters.LastAccess
+                     | NotifyFilters.LastWrite
+                     | NotifyFilters.Size;
+
+            fileWatcher.Changed += OnChanged;
+            fileWatcher.Created += OnCreated;
+            fileWatcher.Deleted += OnDeleted;
+            fileWatcher.Renamed += OnRenamed;
+
+            fileWatcher.Filter = "*.txt";
+            fileWatcher.EnableRaisingEvents = true;
+
+            communicator.PrintMenu();
+            communicator.OpenMenu(this, Console.ReadLine());
+        }
+
+
+        #region TRACKMETHODS
+        private void OnChanged(object sender, FileSystemEventArgs e)
+        {
+            if (e.ChangeType != WatcherChangeTypes.Changed)
+            {
+                return;
+            }
+
+            FileAction change = new Change(path,e.FullPath, FilePastPath[e.FullPath]);
+
+            change.Do();
+
+            Log.Add((DateTime.Now, change));
+        }
+
+
+        private void OnCreated(object sender, FileSystemEventArgs e)
+        {
+            FileAction create = new Creation(e.FullPath);
+
+            FilePastPath.Add(e.FullPath, e.FullPath);
+
+            Log.Add((DateTime.Now, create));
+        }
+
+
+        private void OnDeleted(object sender, FileSystemEventArgs e)
+        {
+            FileAction delet = new Deletion(e.FullPath, FilePastPath[e.FullPath]);
+
+            Log.Add((DateTime.Now, delet));
+        }
+
+
+        private void OnRenamed(object sender, RenamedEventArgs e)
+        {
+            FileAction rename = new Renaming(e.OldFullPath, e.FullPath);
+
+            Log.Add((DateTime.Now, rename));
+        }
```

Comment:
You've created a class hierarchy for actions with these methods, but you don't use inheritance here - you call Do only in one, FilePastPath in the other. No use to inheritance here, the code isn't universalized